app/src/main/java/ch/epfl/sdp/peakar/utils/CameraUtilities.java

```diff
@@ -0,0 +1,111 @@
+package ch.epfl.sdp.peakar.utils;
+
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.hardware.camera2.CameraAccessException;
+import android.hardware.camera2.CameraCharacteristics;
+import android.hardware.camera2.CameraManager;
+import android.util.SizeF;
+
+import androidx.camera.core.AspectRatio;
+import androidx.core.util.Pair;
+
+/**
+ * Utility class for the Camera Activity, contains utility methods for the camera-preview and compass
+ */
+public final class CameraUtilities {
+
+    /**
+     * Calculate the aspect ratio of the display in function of the width and height of the screen
+     * @param width width of the preview in Pixels
+     * @param height height of the preview in Pixels
+     * @return Aspect ratio of the phone
+     */
+    public static int aspectRatio(int width, int height){
```

Comment:
What happens if the actual aspect ratio of the phone is not near this ratio? For example in the case of 21:9?

Reply:
Then it will return the nearest aspectRatio which is 16:9. CameraX only supports 4:3 and 16:9 for now but the camera will still be displayed in fullscreen mode but the picture that is taken without the UI will be in 16:9 aspect ratio.

Reply:
Okay nice thanks!